hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/SchedulePreTransactionHandlerImpl.java

```diff
@@ -75,24 +78,54 @@ public ScheduleTransactionMetadata preHandleCreateSchedule(
         // to see
         // if provided payer is same as payer in the inner transaction.
 
-        final var innerMeta = preHandleInnerTxn(scheduledTxn, payerForNested, dispatcher);
+        final var innerMeta = preHandleScheduledTxn(scheduledTxn, payerForNested, dispatcher);
         meta.scheduledMeta(innerMeta);
         return meta.build();
     }
 
-    private TransactionMetadata preHandleInnerTxn(
+    @Override
+    public ScheduleTransactionMetadata preHandleSignSchedule(
+            final TransactionBody txn,
+            final AccountID payer,
+            final PreHandleDispatcher dispatcher) {
+        final var op = txn.getScheduleSign();
+        final var id = op.getScheduleID();
+
+        final var scheduleLookupResult = scheduleStore.get(id);
+        if (scheduleLookupResult.isEmpty()) {
+            return new InvalidTransactionMetadata(txn, payer, INVALID_SCHEDULE_ID);
+        }
+
+        final var meta =
+                new ScheduleSigTransactionMetadataBuilder(keyLookup)
+                        .txnBody(txn)
+                        .payerKeyFor(payer);
+
+        final var scheduledTxn = scheduleLookupResult.get().scheduledTxn();
+        final var optionalPayer = scheduleLookupResult.get().designatedPayer();
+        final var payerForNested =
+                optionalPayer.orElse(scheduledTxn.getTransactionID().getAccountID());
+
+        final var innerMeta = preHandleScheduledTxn(scheduledTxn, payerForNested, dispatcher);
+        meta.scheduledMeta(innerMeta);
+        return meta.build();
+    }
+
+    private TransactionMetadata preHandleScheduledTxn(
             final TransactionBody scheduledTxn,
             final AccountID payerForNested,
             PreHandleDispatcher dispatcher) {
         final HederaFunctionality scheduledFunction;
         try {
             scheduledFunction = functionOf(scheduledTxn);
         } catch (UnknownHederaFunctionality ex) {
-            return failedMeta(SCHEDULED_TRANSACTION_NOT_IN_WHITELIST, scheduledTxn, payerForNested);
+            return new InvalidTransactionMetadata(
```

Comment:
I wonder, if returning `InvalidTransactionMetadata` is the right thing here. The workflow will simply continue and publish the transaction. But this looks like an error, which will not disappear. Therefore cancelling the workflow and returning an error message seems more appropriate.

Reply:
I agreed. May be in handle, we can just abort with same error if the response is one of these types.